### ISSUE TITLE
"Set as wallpaper" option in overflow menu

### ISSUE THREAD

**Issue by stuartpb** [markusfisch/ShaderEditor#7]
This is possibly my most-used feature in the app, and having to go into Preferences to change it is a total pain. I'd rather just open my shader, then choose "Set as Wallpaper" in the overflow menu. (If I can figure out how, I might submit a PR for this - I'm pretty allergic to native Android/Java development.)

**Comment by stuartpb** [comment]
Also, when implementing this, I would change the current "Choose a shader" dialog to only come up when hitting the "Settings..." option in the home screen Wallpaper chooser (or - at some point in the future - implement a gallery).

**Comment by markusfisch** [comment]
Yes, I see, it's annoying having to go to the preferences for that. Shouldn't be too hard to implement. Will look into it as soon as I got the time.

**Comment by stuartpb** [comment]
Also, there seems to be a bug in the wallpaper-setting logic, where if I set the shader I've already set as my wallpaper to the wallpaper, it doesn't update the wallpaper (and it doesn't update the wallpaper when the shader is recompiled/saved) - lost about 15 minutes to debugging a shader that wasn't broken, because it looked my fixes weren't working.

**Comment by markusfisch** [comment]
Oh, sorry for that. I ran into this lately too and it's not really a bug - it's bad design. At the moment, the shader is only updated when the [preference does change](https://github.com/markusfisch/ShaderEditor/blob/master/src/de/markusfisch/android/shadereditor/ShaderWallpaperService.java#L46).

Embarrassing. Will fix.

**Comment by fekga** [comment]
The wallpaper updates for me when you press the power button (the screen
goes black) and unlock the screen again.

**Comment by markusfisch** [comment]
Yes, good tip!

That's because the wallpaper engine is created anew when the screen is shut off and turned on again. Quite a hack :)

**Comment by markusfisch** [comment]
Okay, I just pushed a [fix](https://github.com/markusfisch/ShaderEditor/commit/55f093da6af8b8e45f283e644f4b141033599a9f). There's now a menu entry to update the shader in the live wallpaper.

I didn't upload a new version into Play yet because I'm planing to do more changes over the next days.

Also, I refactored the project to a more mature structure and converted the project to gradle. Mainly to prepare for Material Design and a more up-to-date UI using a drawer maybe.
